marine/marine.py

```diff
@@ -186,11 +184,12 @@ def validate_fields(
         return bool(self._marine.validate_fields(fields_c_arr, fields_len))
 
     @staticmethod
-    def _parse_output(output: str) -> List[Optional[str]]:
+    def _parse_output(output: List[str], length: int) -> List[Optional[str]]:
         # TODO: this is a bottleneck. Find a better way to provide output from the c code
-        f = StringIO(output)
-        csv_parsed_output = next(csv.reader(f, delimiter="\t", quotechar='"'), [""])
-        return [value or None for value in csv_parsed_output]
+        return (
+            output[i].decode("utf-8") if output[i] is not None else None
```

Comment:
Also `str` has no `decode` in python3, so that should probably be `bytes` anyway

Reply:
Discussed above